crates/cheatnet/tests/starknet/forking.rs

```diff
@@ -0,0 +1,213 @@
+use crate::common::deploy_contract;
+use crate::common::state::create_cheatnet_fork_state;
+use crate::{assert_error, assert_success};
+use blockifier::state::errors::StateError;
+use cairo_felt::Felt252;
+use camino::Utf8PathBuf;
+use cheatnet::cheatcodes::{CheatcodeError, EnhancedHintError};
+use cheatnet::constants::build_testing_state;
+use cheatnet::conversions::{
+    class_hash_from_felt, contract_address_from_felt, felt_selector_from_name,
+};
+use cheatnet::forking::state::ForkStateReader;
+use cheatnet::forking::worker::Worker;
+use cheatnet::rpc::call_contract;
+use cheatnet::state::CustomStateReader;
+use cheatnet::CheatnetState;
+use num_bigint::BigUint;
+use starknet::core::types::BlockId;
+use std::str::FromStr;
+
+#[test]
+fn fork_simple() {
+    let mut state = create_cheatnet_fork_state();
+
+    let contract_address = contract_address_from_felt(&Felt252::from(
+        BigUint::from_str(
+            "3216637956526895219277698311134811322769343974163380838558193911733621219342",
+        )
+        .unwrap(),
+    ));
+
+    let selector = felt_selector_from_name("get_balance");
+    let output = call_contract(&contract_address, &selector, &[], &mut state).unwrap();
+    assert_success!(output, vec![Felt252::from(2)]);
+
+    let selector = felt_selector_from_name("increase_balance");
+    call_contract(
+        &contract_address,
+        &selector,
+        &[Felt252::from(100)],
+        &mut state,
+    )
+    .unwrap();
+
+    let selector = felt_selector_from_name("get_balance");
+    let output = call_contract(&contract_address, &selector, &[], &mut state).unwrap();
+    assert_success!(output, vec![Felt252::from(102)]);
+}
+
+#[test]
+fn try_calling_nonexistent_contract() {
+    let mut state = create_cheatnet_fork_state();
+
+    let contract_address = contract_address_from_felt(&Felt252::from(1));
+    let selector = felt_selector_from_name("get_balance");
+
+    let output = call_contract(&contract_address, &selector, &[], &mut state).unwrap();
+    assert_error!(
+        output,
+        "Contract not deployed at address: 0x0000000000000000000000000000000000000000000000000000000000000001"
+    );
+}
+
+#[test]
+fn try_deploying_undeclared_class() {
+    let mut state = create_cheatnet_fork_state();
+
+    let class_hash = class_hash_from_felt(&Felt252::from(1));
+
+    assert!(match state.deploy(&class_hash, &[]) {
+        Err(CheatcodeError::Unrecoverable(EnhancedHintError::State(
+            StateError::UndeclaredClassHash(undeclared_class_hash),
+        ))) => undeclared_class_hash == class_hash,
+        _ => false,
+    });
+}
+
+#[test]
+fn test_forking_at_block_number() {
```

Comment:
Do I understand correctly that you manually checked the block number of the block that contract with the address was included in?

Reply:
Yup